console-subscriber/src/aggregator/mod.rs

```diff
@@ -769,50 +783,105 @@ impl Aggregator {
                 self.new_poll_ops.push(poll_op);
             }
 
-            Event::StateUpdate {
-                resource_id,
-                update,
-                ..
-            } => {
-                let resource_id = self.ids.id_for(resource_id);
-                if let Some(mut stats) = self.resource_stats.update(&resource_id) {
-                    let field_name = match update.field.name.clone() {
-                        Some(name) => name,
+            Event::StateUpdate { target, update, .. } => {
+                let target_id = self.ids.id_for(target.target_id);
+                let mut targets_to_update = vec![(target_id, target.target_type.clone())];
+
+                fn update_entry(
+                    mut e: std::collections::hash_map::Entry<'_, FieldKey, Attribute>,
+                    upd: &AttributeUpdate,
+                ) {
+                    match e {
+                        Entry::Occupied(ref mut attr) => {
+                            update_attribute(attr.get_mut(), upd);
+                        }
+                        Entry::Vacant(attr) => {
+                            attr.insert(upd.clone().into());
+                        }
+                    }
```

Comment:
take it or leave it: this could also be written as:
```suggestion
                    e
                        .and_modify(|attr| update_attribute(attr.get_mut(), upd))
                        .or_insert_with(|| upd.clone().into());
```